pkg/tests/ossm/prometheus_scoped_scraping_test.go

```diff
@@ -0,0 +1,134 @@
+package ossm
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+
+	"github.com/maistra/maistra-test-tool/pkg/util/check/common"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/require"
+	"github.com/maistra/maistra-test-tool/pkg/util/env"
+	"github.com/maistra/maistra-test-tool/pkg/util/hack"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/shell"
+	"github.com/maistra/maistra-test-tool/pkg/util/test"
+)
+
+func TestOperatorCanUpdatePrometheusConfigMap(t *testing.T) {
+	test.NewTest(t).Groups(test.Full).Run(func(t test.TestHelper) {
+		hack.DisableLogrusForThisTest(t)
+
+		t.Log("This test checks if the operator can update Prometheus ConfigMap when the SMMR is updated")
+
+		meshNamespace := env.GetDefaultMeshNamespace()
+		defaultSMMR := shell.Executef(t, "oc -n %s get smmr default -o jsonpath='{.metadata.annotations.kubectl\\.kubernetes\\.io/last-applied-configuration}'", meshNamespace)
+
+		t.Cleanup(func() {
+			oc.ApplyString(t, meshNamespace, defaultSMMR)
+		})
+
+		t.LogStepf("Delete current SMMR %s", defaultSMMR)
+		oc.DeleteFromString(t, meshNamespace, defaultSMMR)
+
+		getPrometheusConfigCmd := fmt.Sprintf("oc -n %s get configmap prometheus -o jsonpath='{.data.prometheus\\.yml}'", meshNamespace)
+
+		t.NewSubTest("when creating a SMMR").Run(func(t test.TestHelper) {
+			ns := generateNamespace()
```

Comment:
You can use this to generate namespaces names to avoid to have a custom func in the test case: 
`namespaces := util.GenerateStrings("test-", 50)`

Reply:
`util.GenerateStrings` is not what I am after. `generateNamespace` generates a unique long string, which will be used as namespace name. When checking for if the Prometheus configuration is correctly updated, we can just `grep` the unique string in the configuration.